tests/test_core.py

```diff
@@ -0,0 +1,332 @@
+
+    emb1, ts1 = torchopenl3.get_audio_embedding(
+        audio,
+        sr,
+        input_repr="mel128",
+        content_type="env",
+        embedding_size=512,
+        center=True,
+        hop_size=hop_size,
+        verbose=True,
+    )
+    emb1, ts1 = to_numpy(emb1), to_numpy(ts1)
+    assert np.all(np.abs(np.diff(ts1) - hop_size) < tol)
+    assert emb1.shape[1] == 512
+    assert not np.any(np.isnan(emb1))
+
+    emb1, ts1 = torchopenl3.get_audio_embedding(
+        audio,
+        sr,
+        input_repr="mel128",
+        content_type="env",
+        embedding_size=6144,
+        center=True,
+        hop_size=hop_size,
+        verbose=True,
+    )
+    emb1, ts1 = to_numpy(emb1), to_numpy(ts1)
+    assert np.all(np.abs(np.diff(ts1) - hop_size) < tol)
+    assert emb1.shape[1] == 6144
+    assert not np.any(np.isnan(emb1))
+
+    emb1, ts1 = torchopenl3.get_audio_embedding(
+        audio,
+        sr,
+        input_repr="linear",
+        content_type="env",
+        embedding_size=6144,
+        center=True,
+        hop_size=hop_size,
+        verbose=True,
+    )
+    emb1, ts1 = to_numpy(emb1), to_numpy(ts1)
+    assert np.all(np.abs(np.diff(ts1) - hop_size) < tol)
+    assert emb1.shape[1] == 6144
+    assert not np.any(np.isnan(emb1))
+
+    emb1, ts1 = torchopenl3.get_audio_embedding(
+        audio,
+        sr,
+        input_repr="linear",
+        content_type="env",
+        embedding_size=512,
+        center=True,
+        hop_size=hop_size,
+        verbose=True,
+    )
+    emb1, ts1 = to_numpy(emb1), to_numpy(ts1)
+    assert np.all(np.abs(np.diff(ts1) - hop_size) < tol)
+    assert emb1.shape[1] == 512
+    assert not np.any(np.isnan(emb1))
+
+    # Make sure we can load a model and pass it in
+    model = torchopenl3.models.load_audio_embedding_model("linear", "env", 512)
+    emb1load, ts1load = torchopenl3.get_audio_embedding(
+        audio, sr, model=model, center=True, hop_size=hop_size, verbose=True
+    )
+    emb1load, ts1load = to_numpy(emb1load), to_numpy(ts1load)
+    assert np.all(np.abs(emb1load - emb1) < tol)
+    assert np.all(np.abs(ts1load - ts1) < tol)
+
+    # Make sure that the embeddings are approximately the same with mono and stereo
+    audio, sr = sf.read(filenames["CHIRP_STEREO_PATH"])
+    emb2, _ = torchopenl3.get_audio_embedding(
+        audio, sr, model=model, center=True, hop_size=0.1, verbose=True
+    )
+    emb2 = to_numpy(emb2)
+    # assert np.all(np.abs(emb1 - emb2) < tol)
+    # assert np.all(np.abs(ts1 - ts2) < tol)
+    assert not np.any(np.isnan(emb2))
+
+    # Make sure that the embeddings are approximately the same if we resample the audio
+    audio, sr = sf.read(filenames["CHIRP_44K_PATH"])
+    emb3, _ = torchopenl3.get_audio_embedding(
+        audio, sr, model=model, center=True, hop_size=0.1, verbose=True
+    )
+    emb3 = to_numpy(emb3)
+    # assert np.all(np.abs(emb1 - emb3) < tol)
+    # assert np.all(np.abs(ts1 - ts3) < tol)
```

Comment:
same with the resampling.

Reply:
We can do this. First I thought to check if test case are passing for others or not.

Reply:
sure. i think that will work